### ISSUE TITLE
Apply BIDS-like naming convention for data files and reorganize them slightly

### ISSUE THREAD

**Issue by tsalo** [neuroquery/neuroquery_data#3]
For NiMARE/Neurostore conversion, I think it would be useful to reorganize the files a bit. I have four proposed sets of changes:

First, it would be great to adopt some key-value pairs in the `npz` filenames, in a pseudo-BIDS format. Perhaps the following entities?

- source: The origin of the counts/tfidf values, such as "title", "abstract", or "combined".
- vocab: The set of terms. Maybe corpus would work here too?
- type: Just "count" vs. "tfidf".
- version: Unused at the moment (could be 1 I guess), but this would be particularly useful for Neurosynth.

So the files might be something like `source-abstract_vocab-standard_type-count_neuroquery.npz`.

Second, the vocabulary txt files could adopt a reduced version: `vocab-standard_neuroquery.txt`, `vocab-expanded_neuroquery.txt`, and `vocab-combined_neuroquery.txt`.

Third, since the PMIDs should be shared across all files in the dataset (unless I'm mistaken), then that could be moved to the top level. I think the same goes for the corpus metadata file.

Finally, I think the data could all go in `training_data` or a general `data` folder.

I would then try to mirror the naming convention in `neurosynth-data` (see https://github.com/neurosynth/neurosynth-data/issues/5), update NiMARE's ingestion function and fetchers (https://github.com/neurostuff/NiMARE/issues/522), and finally update the neurostore ingestion function as well.

**Comment by jeromedockes** [comment]
thanks! all these changes sound good.
I would only keep the neuroquery_model.zip as it is, as it is a pre-trained model so something different than the training data, and it is downloaded and used by the `neuroquery` package

**Comment by jeromedockes** [comment]
this:

https://github.com/neurostuff/NiMARE/issues/522#issuecomment-867169369

also still needs to be done (removing the few studies in training_data that are not used in neuroquery_model)

**Comment by adelavega** [comment]
LGTM. I suppose since there's no actual standard to adhere to there's not much to worry about it, but it will make it easier to know what we're looking at.

Only concern is that it might possibly break some stuff in neurosynth python library but either way that is very deprecated at this point.

**Comment by tsalo** [comment]
Thanks @jeromedockes and @adelavega. I took a crack at it in https://github.com/neurosynth/neurosynth-data/pull/6. One thing to note- once I started actually reorganizing the Neurosynth files, I ended up deciding to change the format slightly. Namely, I moved the database to a new "data" entity at the beginning and using the suffix for the data type (i.e., "database", "features", "vocab", and "ids"). Below is a screenshot of the list of version-7 files for Neurosynth. Please let me know if you see any issues with the updated convention:

![image](https://user-images.githubusercontent.com/8228902/124319245-eec54a00-db47-11eb-8537-a57f69ff2696.png)

**Comment by adelavega** [comment]
LGTM. @jdkent do you want to comment on this?

**Comment by jeromedockes** [comment]
btw, .npz was chosen just because it was convenient to use with the dependencies neuroquery already had but it has the disadvantage of being specific to scipy -- if you have another suggestion for serializing sparse matrices that can be changed as well

**Comment by tsalo** [comment]
@jeromedockes That's a good point. I never work with sparse matrices, so I don't actually know anything about the different options. I don't have a problem with using scipy though. 

For neurosynth/neurosynth-data#6 I used [this notebook](https://github.com/tsalo/misc-notebooks/blob/369b9c2ec417464fa8202080a77abee8685b926e/reorganize_neurosynth_data.ipynb). I used `scipy.sparse.csc_matrix` to convert to a sparse matrix, but I realized that I don't know what specific format you used...

**Comment by tsalo** [comment]
I'm thinking a better entity order might be data, version, vocab, source, type. That way, related files will end up closer.

**Comment by jeromedockes** [comment]
"features" and "database" are not very informative IMO... do you thing we could name them "textfeatures" or "vectorizedtext" and "coordinates", or something like that?

**Comment by tsalo** [comment]
I didn't change Neurosynth's database files at all, beyond gzipping them, so they contain both coordinates and metadata. My thinking is that those files should have everything for the database outside of annotations.

Not sure what the best replacement for "features" is. Hypothetically, one could store a manually-annotated dataset in this format, in which case the annotations wouldn't come from text, but I don't know if that's a likely enough scenario to make excluding "text" from the name necessary. I prefer "textfeatures" over "vectorizedtext", though, since "vectorizedtext" feels a little specific.

**Comment by jeromedockes** [comment]
> My thinking is that those files should have everything for the database outside of annotations.

so the studies' metadata is duplicated for each coordinate?

**Comment by jeromedockes** [comment]
> Hypothetically, one could store a manually-annotated dataset in this format, in which case the annotations wouldn't come from text

wouldn't that be a different kind of file then? it probably wouldn't be sparse, there wouldn't be a `vocab-`, ...

**Comment by tsalo** [comment]
I think the same file format is still valid for manual annotations. You're right that it probably wouldn't be sparse, although it definitely _could_ be. In those kinds of annotations, I'd expect binary values, and the amount of zeros would depend on the scope of the ontology. If you pulled BrainMap and converted it to this format, I'd expect a fairly sparse matrix. I'd probably label a manually-annotated features file `vocab-cogpo`, `vocab-cogat`, or `vocab-[author]`, depending on the ontology used. 

>so the studies' metadata is duplicated for each coordinate?

It is, but the database file isn't that big compared to the feature files, so I'm fine with the duplication. We _could_ separate any study-level information out, but then we're talking about managing, at minimum, four files instead of three.

**Comment by jeromedockes** [comment]
> I think the same file format is still valid for manual annotations.

I see, in that case if you prefer I'm ok with keeping "features" even though it's not very informative. the "_database" suffix OTOH contains zero information so we might as well not have it at all

 > We could separate any study-level information out, but then we're talking about managing, at minimum, four files instead of three.

yes that is how it was in neuroquery so far. it's not only about the size but avoiding the duplication makes it possible to have a meaningful index and makes it clearer that there is only one information. also, although it isn't the case now, text features or annotations of studies that don't contain any coordinates could be useful for example to compute term similarities, topic models etc and we would still want their metadata. this is not a very strong opinion on my part but I find distributing the join of the tables instead of their normal forms a bit surprising

**Comment by tsalo** [comment]
Okay, so we can have `coordinates.tsv.gz`, with, at minimum, `id`, `table_id`, `x`, `y`, and `z` columns. Additional columns that'll be in the Neurosynth data would be `peak_id` and `table_num`. For NeuroQuery, the only additional column would be `table_name`, right?

In `metadata.tsv.gz` files, Neurosynth will have the columns `id`, `doi`, `space`, `title`, `authors`, `year`, and `journal`. NeuroQuery will have `id`, `title`, and `pubmed_url`, I think?

Does that work?

Oh and `metadata.tsv.gz` would probably have the same entities as `coordinates.tsv.gz` (`data` and `version`).

**Comment by tsalo** [comment]
@jeromedockes do you have any concerns about the proposed changes? I can incorporate them fairly easily into the Neurosynth and NiMARE PRs, but I want to make sure they're good with you first.

**Comment by jeromedockes** [comment]
sorry for the late reply! yes that sounds good, thanks. so IIUC on the NeuroQuery side what needs to change is renaming the files, changing from csv to tsv, and gzipping them?

**Comment by tsalo** [comment]
I think so. Some of the NeuroQuery files might have "pmid" instead of "id" though.

**Comment by tsalo** [comment]
Updated file list (just version 7):

![image](https://user-images.githubusercontent.com/8228902/128747940-ac675e9e-760c-41ed-a403-0484f3eb634c.png)

Two small notes:

1. I used `na_rep="n/a"` to follow BIDS convention.
1. I used `line_terminator="\n"` just to be safe. I think that's also BIDS convention.
